components/kserve/servicemesh_setup.go

```diff
@@ -1,47 +1,59 @@
 package kserve
 
 import (
+	"fmt"
 	"path"
 
 	operatorv1 "github.com/openshift/api/operator/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
 )
 
-func (k *Kserve) configureServiceMesh(dscispec *dsciv1.DSCInitializationSpec) error {
+func (k *Kserve) configureServiceMesh(c client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
 	if dscispec.ServiceMesh.ManagementState == operatorv1.Managed && k.GetManagementState() == operatorv1.Managed {
-		serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures())
+		serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures(c))
 		return serviceMeshInitializer.Apply()
 	}
 	if dscispec.ServiceMesh.ManagementState == operatorv1.Unmanaged && k.GetManagementState() == operatorv1.Managed {
 		return nil
 	}
 
-	return k.removeServiceMeshConfigurations(dscispec)
+	return k.removeServiceMeshConfigurations(c, dscispec)
 }
 
-func (k *Kserve) removeServiceMeshConfigurations(dscispec *dsciv1.DSCInitializationSpec) error {
-	serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures())
+func (k *Kserve) removeServiceMeshConfigurations(cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	serviceMeshInitializer := feature.ComponentFeaturesHandler(k.GetComponentName(), dscispec, k.defineServiceMeshFeatures(cli))
 	return serviceMeshInitializer.Delete()
 }
 
-func (k *Kserve) defineServiceMeshFeatures() feature.FeaturesProvider {
+func (k *Kserve) defineServiceMeshFeatures(cli client.Client) feature.FeaturesProvider {
 	return func(handler *feature.FeaturesHandler) error {
-		kserveExtAuthzErr := feature.CreateFeature("kserve-external-authz").
-			For(handler).
-			Manifests(
-				path.Join(feature.KServeDir, "activator-envoyfilter.tmpl"),
-				path.Join(feature.KServeDir, "envoy-oauth-temp-fix.tmpl"),
-				path.Join(feature.KServeDir, "kserve-predictor-authorizationpolicy.tmpl"),
-				path.Join(feature.KServeDir, "z-migrations"),
-			).
-			WithData(servicemesh.ClusterDetails).
-			Load()
+		authorinoInstalled, err := deploy.ClusterSubscriptionExists(cli, "authorino-operator")
+		if err != nil {
+			return fmt.Errorf("failed to list subscriptions %w", err)
+		}
```

Comment:
This is redundant to the check we have in DSCI, I will think about how to make it centralized, but for now, I would stick to this simplistic approach.  Any objections?